### PR TITLE
Update test OS matrix for release/3.1

### PR DESCRIPTION
**Pull request by bartonjs**
* Run servicing PRs as fullMatrix: true
* Refresh all docker images to their latest tag
* Added Alpine.311 (amd64)
* Added Fedora.30
* Removed Ubuntu.1904
* Added Ubuntu.1910
* Added OSX.1015

### CODE REVIEW

File: eng/pipelines/linux.yml [hunk 1]
```diff
@@ -123,15 +123,15 @@ stages:
           - redhatHelixQueue: RedHat.6.Amd64.Open
 
           - ${{ if eq(parameters.fullMatrix, 'false') }}:
-            - linuxDefaultQueues: Centos.7.Amd64.Open+RedHat.7.Amd64.Open+Debian.9.Amd64.Open+Ubuntu.1604.Amd64.Open+Ubuntu.1804.Amd64.Open+SLES.15.Amd64.Open+\(Fedora.29.Amd64.Open\)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-29-09ca40b-20190508143249
-            - alpineQueues: \(Alpine.39.Amd64.Open\)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.9-helix-09ca40b-20190620160617
+            - linuxDefaultQueues: Centos.7.Amd64.Open+RedHat.7.Amd64.Open+Debian.9.Amd64.Open+Ubuntu.1604.Amd64.Open+Ubuntu.1804.Amd64.Open+SLES.15.Amd64.Open+\(Fedora.29.Amd64.Open\)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-29-helix-a12566d-20191210224553
```
Comment (safern) [inline eng/pipelines/linux.yml:126]: Do we need to update these? Or can we just remove them as fullMatrix is now always true?

**Review by safern — APPROVED** [review]

**Review by ViktorHofer — APPROVED** [review]
I agree with @safern, if we now always run on the full matrix we should clean the conditions up and remove the not full matrix sections.

File: eng/pipelines/linux.yml (hunk 1 above)
Comment (bartonjs) [reply eng/pipelines/linux.yml:126]: I did that at first; but maintaining this list seems easier than putting the whole thing back if we decide that always doing fullMatrix doesn't go so well.

Comment (ViktorHofer) [reply eng/pipelines/linux.yml:126]: Based on the low number of PRs in release/3.1 this should be fine. I can't think of any reason why we would need to switch back.

**Comment by bartonjs** [comment]
@wfurt System.Net.NetworkInformation.Tests.NetworkInterfaceBasicTest.BasicTest_AccessInstanceProperties_NoExceptions_Bsd looks like it's failing intermittently (10.14, 10.15).  Should we disable the test in release/3.1? I see it has been disabled in master (https://github.com/dotnet/runtime/issues/1332)...

**Comment by wfurt** [comment]
yes, I think we should @bartonjs. I could not reproduce it locally but I see no point of leaving unstable test enabled in the release branch. Alternatively, we can update the test in the release branch to not assert on known interface state.

**Comment by bartonjs** [comment]
Alrighty, the only failing test now is that regex testing hit some OOMs on Alpine.38.  If it keeps that up we might need to trim back the tests; but it certainly hasn't been a heavy hitter during the rest of this PR.

**Comment by tarekgh** [comment]
The Globalization tests change LGTM.

**Comment by danmoseley** [comment]
The regex OOM's on Alpine for reference were
```
Match_SpecialUnicodeCharacters_enUS
Match_ExcessPrefix
Match_Timeout_Throws
Match_SpecialUnicodeCharacters_Invariant
```
the stack looks like perhaps the fork itself is failing, which would not be a regex issue (?)
```
   at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec) in /_/src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs:line 543
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo) in /_/src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs:line 453
   at System.Diagnostics.Process.Start() in /_/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs:line 1222
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo) in /_/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs:line 1265
   at Microsoft.DotNet.RemoteExecutor.RemoteExecutor.Invoke(MethodInfo method, String[] args, RemoteInvokeOptions options, Boolean pasteArguments) in /_/src/Microsoft.DotNet.RemoteExecutor/src/RemoteExecutor.cs:line 331
   at Microsoft.DotNet.RemoteExecutor.RemoteExecutor.Invoke(Func`1 method, RemoteInvokeOptions options) in /_/src/Microsoft.DotNet.RemoteExecutor/src/RemoteExecutor.cs:line 161
   a
```
No action required here - we'll see if this becomes an issue in future PR's.

**Comment by danmoseley** [comment]
2.1 change was already merged: https://github.com/dotnet/corefx/pull/42838

**Comment by danmoseley** [comment]
@bartonjs I forgot these are actually just "tell" mode -- we do not need tactics approval. As usual @Anipik  will do the merge at the right time

**Pull request merged**